Missing header to compile src/libfetchers/fetchers.cc

```
> nix develop
> configurePhase
> buildPhase
...snip...

FAILED: [code=1] src/libfetchers/libnixfetchers.2.35.0.dylib.p/fetchers.cc.o
clang++ -Isrc/libfetchers/libnixfetchers.2.35.0.dylib.p -Isrc/libfetchers -I../src/libfetchers -Isrc/libfetchers/include -I../src/libfetchers/include -Isrc/libutil/include -I../src/libutil/include -I../src/libutil/widecharwidth -Isrc/libutil/unix -I../src/libutil/unix -Isrc/libutil/unix/include -I../src/libutil/unix/include -Isrc/libstore/include -I../src/libstore/include -Isrc/libstore/unix/include -I../src/libstore/unix/include -I/nix/store/69zmjsczc9ra332xk6ypzkx00ra63hj6-nlohmann_json-3.12.0/include -I/nix/store/v54bmgm7y002yvg821xwlskqmqkqpwfn-libgit2-1.9.2-dev/include -I/nix/store/34v5aih76hsgri5v7ch9msqfimihmhpr-libarchive-3.8.4-dev/include -fdiagnostics-color=always -fpch-instantiate-templates -include-pch src/libfetchers/libnixfetchers.2.35.0.dylib.p/precompiled-headers.hh.pch -Wall -Winvalid-pch -std=c++23 -O2 -g -Wdeprecated-copy -Werror=suggest-override -Werror=switch -Werror=switch-enum -Werror=undef -Werror=unused-result -Werror=sign-compare -Werror=return-type -Werror=non-virtual-dtor -Wignored-qualifiers -Wimplicit-fallthrough -Wno-deprecated-declarations -fpch-instantiate-templates -Werror=c99-designator -std=c++23 -std=c++23 -MD -MQ src/libfetchers/libnixfetchers.2.35.0.dylib.p/fetchers.cc.o -MF src/libfetchers/libnixfetchers.2.35.0.dylib.p/fetchers.cc.o.d -o src/libfetchers/libnixfetchers.2.35.0.dylib.p/fetchers.cc.o -c ../src/libfetchers/fetchers.cc
../src/libfetchers/fetchers.cc:361:27: error: no member named 'sleep_for' in namespace 'std::this_thread'
  361 |         std::this_thread::sleep_for(std::chrono::seconds(1));
      |                           ^~~~~~~~~
1 error generated.
```

--- a/src/libfetchers/fetchers.cc
+++ b/src/libfetchers/fetchers.cc
@@ -11,6 +11,7 @@
 #include "nix/store/pathlocks.hh"
 #include "nix/util/environment-variables.hh"
 
+#include <thread>
 #include <nlohmann/json.hpp>
 
 namespace nix::fetchers {

